cmd/lambda/leases/delete_test.go

```diff
@@ -108,6 +113,102 @@ func TestDeleteController_Call(t *testing.T) {
 
 }
 
+func TestDeleteByID(t *testing.T) {
+
+	type response struct {
+		StatusCode int
+		Body       string
+	}
+	tests := []struct {
+		name               string
+		expResp            response
+		leaseID            string
+		getLease           *db.Lease
+		getErr             error
+		transitiionedLease *db.Lease
+		transitionErr      error
+	}{
+		{
+			name:    "successful delete",
+			leaseID: "abc123",
+			expResp: response{
+				StatusCode: 200,
+				Body:       "{\"accountId\":\"123456789012\",\"principalId\":\"principal\",\"id\":\"abc123\",\"leaseStatus\":\"Inactive\",\"leaseStatusReason\":\"\",\"createdOn\":0,\"lastModifiedOn\":0,\"budgetAmount\":0,\"budgetCurrency\":\"\",\"budgetNotificationEmails\":null,\"leaseStatusModifiedOn\":0,\"expiresOn\":0,\"metadata\":null}\n",
+			},
+			getLease: &db.Lease{
+				ID:          "abc123",
+				LeaseStatus: db.Active,
+				PrincipalID: "principal",
+				AccountID:   "123456789012",
+			},
+			transitiionedLease: &db.Lease{
+				ID:          "abc123",
+				LeaseStatus: db.Inactive,
+				PrincipalID: "principal",
+				AccountID:   "123456789012",
+			},
+			getErr: nil,
+		},
+		{
+			name:    "failure no lease by that ID",
+			leaseID: "abc123",
+			expResp: response{
+				StatusCode: 500,
+				Body:       "{\"error\":{\"code\":\"ServerError\",\"message\":\"Cannot verify if Lease ID \\\"abc123\\\" exists\"}}",
+			},
+			getLease: &db.Lease{
+				ID:          "abc123",
+				LeaseStatus: db.Active,
+				PrincipalID: "principal",
+				AccountID:   "123456789012",
+			},
+			transitiionedLease: &db.Lease{
+				ID:          "abc123",
+				LeaseStatus: db.Inactive,
+				PrincipalID: "principal",
+				AccountID:   "123456789012",
+			},
+			getErr: fmt.Errorf("failure"),
+		},
```

Comment:
Can you please add a test for more than one lease exists with input lease ID and in this case DB error should be expected.

Reply:
I'll add a test but it wouldn't matter here.  This logic is done in the db package.  Which will return an error if the result is less than 1 or if the result is more than 1.

I have added a test to the db package.